src/test/java/com/coupang/marketplace/order/service/OrderServiceTest.java

```diff
@@ -0,0 +1,78 @@
+import com.coupang.marketplace.global.fixture.AddressFixture.*;
+import com.coupang.marketplace.global.fixture.CartFixture.*;
+import com.coupang.marketplace.global.fixture.PaymentFixture.*;
+import com.coupang.marketplace.global.fixture.UserFixture.*;
+import com.coupang.marketplace.order.controller.dto.OrderRequestDto;
+import com.coupang.marketplace.order.repository.OrderRepository;
+import com.coupang.marketplace.payment.repository.PaymentRepository;
+import com.coupang.marketplace.payment.service.PaymentService;
+import com.coupang.marketplace.product.domain.Product;
+import com.coupang.marketplace.product.repository.ProductRepository;
+import com.coupang.marketplace.user.repository.UserRepository;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.times;
+
+@ExtendWith(MockitoExtension.class)
+public class OrderServiceTest {
+
+    @InjectMocks
+    private OrderService orderService;
+
+    @Mock
+    private CartRepository cartRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private AddressRepository addressRepository;
+
+    @Mock
+    private ProductRepository productRepository;
+
+    @DisplayName("장바구니에 담긴 상품을 주문한다.")
+    @Test
+    public void order() {
+//        //given
+//        given(cartRepository.findByUserId(User1.ID)).willReturn(Arrays.asList(Cart1.CART));
+//        given(userRepository.findById(User1.ID)).willReturn(User1.USER);
+//        given(addressRepository.findMainContentByUserId(User1.ID)).willReturn(Address1.CONTENT);
+//        //then
+//        then(cartRepository).should(times(1)).deleteCartProducts(Cart1.CART);
+    }
+
+    @DisplayName("주문을 요청한 상품이 없는 경우 주문에 실패한다.")
+    @Test
+    public void NoProductToOrder() {
+//        //given
+//        final Optional<Cart> notFoundCartProduct = Optional.ofNullable(null);
+//        final Optional<Product> notFoundProduct = Optional.ofNullable(null);
+//        final OrderRequestDto dto = OrderRequestDto.builder()
+//                        .type(Payment1.TYPE)
+//                        .receiverName(User1.NAME)
+//                        .receiverPhone(User1.PHONE)
+//                        .receiverRequest("")
+//                        .build();
+//        given(cartRepository.findByUserId(User1.ID)).willReturn(Arrays.asList());
+//        given(productRepository.findByProductId(notFoundCartProduct.get().getProductId())).willReturn(notFoundProduct);
+//        //then
+//        assertThrows(IllegalArgumentException.class, () -> orderService.order(User1.ID, dto));
```

Comment:
이 부분은 이렇게 쓰면 동작합니다!
이미 유저의 장바구니가 비어있으면 stream 에서 처리할 게 없어서 굳이 notFound부분 처리 안해줘도 되기 때문입니다~!

```suggestion
        //given
        final OrderRequestDto dto = OrderRequestDto.builder()
                        .type(Payment1.TYPE)
                        .receiverName(User1.NAME)
                        .receiverPhone(User1.PHONE)
                        .receiverRequest("")
                        .build();
        given(cartRepository.findByUserId(User1.ID)).willReturn(Arrays.asList());

        //then
        assertThrows(IllegalArgumentException.class, () -> orderService.order(User1.ID, dto));
```

Reply:
아 그렇군요!! 도움주셔서 감사합니다 ㅎㅎ